### PR TITLE
Hyperband oob

### PR DESCRIPTION
**Pull request by donglinjy**
Move hyperband from a separate algorithms plugin [orion.core.hyeprband](https://github.com/Epistimio/orion.algo.hyperband) to an OOB algorithms in Orion core.

### CODE REVIEW

**Comment by donglinjy** [comment]
@bouthilx

File: src/orion/algo/hyperband.py [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+                point = list(point)
+                point[self.fidelity_index] = bracket.rungs[0]['resources']
+                points.append(tuple(point))
+            i += 1
+
+        if not points:
+            raise RuntimeError(
+                'Hyperband keeps sampling already existing points. This should not happen, '
+                'please report this error to '
+                'https://github.com/bouthilx/orion.algo.hyperband/issues')
```
Comment (bouthilx) [inline src/orion/algo/hyperband.py]:
```suggestion
                'https://github.com/Epistimio/orion/issues')
```

File: tests/functional/algos/test_algos.py [hunk 2]
```diff
@@ -12,6 +12,7 @@
 
 config_files = ['random_config.yaml']
 fidelity_config_files = ['random_config.yaml', 'asha_config.yaml']
+hyperband_config = ['hyperband.yaml']
```
Comment (bouthilx) [inline tests/functional/algos/test_algos.py]: Can you add the yaml file to the list above instead so that we don't need to add a test explicitly?

**Comment by bouthilx** [comment]
Documentation will also need to be updated: https://orion.readthedocs.io/en/latest/user/algorithms.html#included-algorithms

**Comment by bouthilx** [comment]
[Add Hyperband to Oríon](https://trello.com/c/fwjQ4gqG/453-add-hyperband-to-or%C3%ADon)

File: src/orion/algo/hyperband.py (hunk 1 above)
Comment (donglinjy) [reply src/orion/algo/hyperband.py]: > Documentation will also need to be updated: https://orion.readthedocs.io/en/latest/user/algorithms.html#included-algorithms
> 
> (here https://github.com/Epistimio/orion/blob/master/docs/src/user/algorithms.rst)

Will add doc for hyperband later when we get some other issues done which may affect how to write the doc.

**Comment by donglinjy** [comment]
> Documentation will also need to be updated: https://orion.readthedocs.io/en/latest/user/algorithms.html#included-algorithms
> 
> (here https://github.com/Epistimio/orion/blob/master/docs/src/user/algorithms.rst)

Will add doc for hyperband later when we get some other issues done which may affect how to write the doc.

**Pull request merged**